### DISCUSSION TITLE
FreshRSS

### DISCUSSION THREAD

**Discussion by JohnShortland** [htpcBeginner/docker-traefik#100] [Ideas]
Has anyone added FreshRSS to this stack and managed to bypass oauth for it, I've been trying but just can't get the Andriod app, readrops to log me in.

**Comment by Cartache** [comment]
This works for me:

 freshrss:
    container_name: freshrss
    image: linuxserver/freshrss:latest
    restart: always
    networks:
      - t2_proxy
    security_opt:
      - no-new-privileges:true
    volumes:
      - $DOCKERDIR/freshrss:/config
    links:
      - mariadb:db
    depends_on:
      - mariadb      
    environment:
      - PUID=$PUID
      - PGID=$PGID
      - TZ=$TZ
    labels:
      - "traefik.enable=true"
      ## HTTP Routers
      - "traefik.http.routers.freshrss-rtr.entrypoints=https"
      - "traefik.http.routers.freshrss-rtr.rule=Host(`freshrss.$DOMAINNAME`)" 
      - "traefik.http.routers.freshrss-rtr.tls=true"
      ## Middlewares
      - "traefik.http.routers.freshrss-rtr.middlewares=chain-noauth@file" 
      ## HTTP Services
      - "traefik.http.routers.freshrss-rtr.service=freshrss-svc"
      - "traefik.http.services.freshrss-svc.loadbalancer.server.port=80"
`

**Comment by robflate** [comment]
Obviously, the above solution will work but it has security implications as it removes OAth completely and exposes the whole FreshRSS instance, relying only on the basic authentication from FreshRSS.

I don't think I managed to bypass OAuth for the API by checking for the presence of the API key in the header. It's been ages since I tried so can't remember how far I got. A quick look at this page suggests you'll need to use Tokens or check the headers for other data.

https://freshrss.github.io/FreshRSS/en/users/06_Mobile_access.html

You could try something like;

```
    labels:
      - "traefik.enable=true"
      ## HTTP Routers
      - "traefik.http.routers.freshrss-rtr.entrypoints=https"
      - "traefik.http.routers.freshrss-rtr.rule=Host(`freshrss.$DOMAINNAME`)"
      - "traefik.http.routers.freshrss-rtr.priority=99"
      ## Middlewares
      - "traefik.http.routers.freshrss-rtr.middlewares=chain-oauth@file"
      ## HTTP Services
      - "traefik.http.routers.freshrss-rtr.service=freshrss-svc"
      - "traefik.http.services.freshrss-svc.loadbalancer.server.port=80"
      ## HTTP Routers Auth Bypass
      - "traefik.http.routers.freshrss-rtr-bypass.entrypoints=https"
      - "traefik.http.routers.freshrss-rtr-bypass.rule=Host(`freshrss.$DOMAINNAME`) && PathPrefix(`/api`)"
      - "traefik.http.routers.freshrss-rtr-bypass.priority=100"
      - "traefik.http.routers.freshrss-rtr-bypass.service=freshrss-svc"
      - "traefik.http.routers.freshrss-rtr-bypass.middlewares=chain-no-auth@file"
```

This should work by only exposing `/api` and keeping OAuth on the rest. Again, it has security implications.

I use [Miniflux](https://miniflux.app) instead.

**Reply by JohnShortland** [reply]
I quite like the look of Miniflux might try that instead.
